Sources/SpelledPitch/PitchSpeller/Wetherfield/SpellingInverter.swift

```diff
@@ -160,24 +207,125 @@ extension SpellingInverter {
     
     // MARK: - Instance Methods
     
+    mutating func partition (_ indices: [Int: Int]) {
+        let adjacencyScheme = GraphScheme<FlowNode<Int>> { edge in
+            switch (edge.a, edge.b) {
+            case let (.internal(a), .internal(b)):
+                return indices[a]! == indices[b]!
```

Comment:
You can get away with

```Swift
return indices[a] == indices[b]
```

as the `Optional` type gets `Equatable` conformance for free.

Reply:
nice